lib/galaxy/model/migrations/alembic/versions_gxy/1d0564b6d7aa_make_api_keys_deleted_non_nullable.py

```diff
@@ -0,0 +1,32 @@
+"""make api_keys deleted column non-nullable
+
+Revision ID: 1d0564b6d7aa
+Revises: caa7742f7bca
+Create Date: 2023-04-19 11:09:46.504245
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "1d0564b6d7aa"
+down_revision = "caa7742f7bca"
+branch_labels = None
+depends_on = None
+
+
+table_name = "api_keys"
+column_name = "deleted"
+
+
+def upgrade():
+    # Update any existing rows with a deleted=NULL value to be False.
+    table = sa.sql.table(table_name, sa.Column(column_name, sa.Boolean(), nullable=True))
+    op.execute(table.update().where(table.c.deleted.is_(None)).values(deleted=False))
```

Comment:
If you do want to alter these you'll have to make it so that only the newest API key is set to `deleted=False` and all older ones are set to `deleted=True`.

Reply:
and of course only if there isn't one that is already valid for that user.

Reply:
```suggestion
    op.execute(table.update().where(table.c.deleted.is_(None)).values(deleted=True))
```

even without the sorting, safer to just have it fail closed.

Reply:
Done in the rebase to keep the migration script in a single commit, thanks!